lib/sidecar.js

```diff
@@ -1390,4 +1412,96 @@ class Engine extends ReadyResource {
   }
 }
 
+class Collector {
+  async sidecar ({ pid = null, name = 'pear-runtime', flag = '--sidecar' } = {}) {
+    const current = Bare.pid
+    const ids = await this.#lookupProcess({ name, flag })
+    const filtered = ids.filter(id => id !== pid && id !== current)
+    const killed = await this.#kill(filtered)
+    return { killed }
+  }
+
+  async #lookupProcess ({ name = null, flag = null } = {}) {
+    if (!name || !flag) throw new Error('Process name or flag are missing')
+
+    const [sh, args] = isWindows
+      ? ['cmd.exe', ['/c', `wmic process where "executablepath like '%${name}%'" get processid, commandline`]]
+      : ['/bin/sh', ['-c', `ps ax | grep -i -- '${name}' | grep -i -- '${flag}'`]]
+
+    return new Promise((resolve, reject) => {
+      const sp = spawn(sh, args)
+      const output = []
```

Comment:
there's no need to buffer the output - send it over the wire immediataley by yielding it from the parent - easiest way is to make it so Collector sidecar returns a stream instead of a promise and then yield the tags { tag: 'stderr, data ... }